src/simulator/synapse.c

```diff
@@ -216,4 +216,72 @@ synapse_clear(Synapse* synapse) {
     
     error:
     return;
-}
\ No newline at end of file
+}
+
+
+internal f32
+synapse_stdp_potentiation_weight_update(u32 interval) {
+    return 0.1f;
+}
+
+internal void
+synapse_stdp_potentiation_update(Synapse* synapse) {
+    check(synapse != NULL, "synapse is NULL");
+
+    u32 out_neuron_spike_time = synapse->out_neuron->last_spike_time;
+    u32 in_neuron_spike_time = synapse->in_neuron->last_spike_time;
+
+    // NOTE: the input neuron should have spiked
+    // TODO: add function that return true if the neuron ever spiked, easier to read
+    if (in_neuron_spike_time == NEURON_INVALID_SPIKE_TIME) return;
+
+    u32 interval_value = out_neuron_spike_time - in_neuron_spike_time;
+    // NOTE: the interval between the spike should be big enough and it should not be 0
+    // TODO: should it be bigger than the synapse delay????????
+    if (interval_value > SYNAPSE_POTENTIATION_INTERVAL ||
+        interval_value == 0)
```

Comment:
Yes should be bigger than the synapse delay in order for the pre synaptic spike to have an effect on the postsynaptic neuron